boards/posix/nrf52_bsim/main.c

```diff
@@ -57,8 +57,6 @@ void posix_exit(int exit_code)
 	exit(inner_main_clean_up(exit_code));
 }
 
-uint global_device_nbr;
```

Comment:
Oh well.. that change just broke a bunch of off-tree tests..
@alxelax Could you please revert get this back asap?

Reply:
Sure, I could. Which tests are you mentioning about? I fixed all places where it has been used in Zephyr repo. All tests passed in the commit checking test plans.

Reply:
See #37891